lexers/s/ssp.go

```diff
@@ -0,0 +1,41 @@
+package s
+
+import (
+	"regexp"
+	"strings"
+
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+	"github.com/alecthomas/chroma/pkg/xml"
+)
+
+var sspAnalyserRe = regexp.MustCompile(`val \w+\s*:`)
+
+// SSP lexer. Lexer for Scalate Server Pages.
+var SSP = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "SSP",
+		Aliases:   []string{"ssp"},
+		Filenames: []string{"*.ssp"},
+		MimeTypes: []string{"application/x-ssp"},
+	},
+	Rules{
+		"root": {},
+	},
+).SetAnalyser(func(text string) float32 {
+	var result float64
```

Comment:
Can't it be declared as float32?

Reply:
I did this first, but the test failed, when all 3 increments (`+=0.6`, `+=0.2`, `+=0.1`) were performed. It did not add up to `0.9`, but to `0.900004`.

This shows again, that arithmetic with float numbers is not safe. That's why for finance applications they strictly use integer. But here we can solve it by using `float64` inside the function. (btw, didn't you encountered some similar case? It felt familiar..)

Reply:
The cases I needed a float64 was because I had to use math.Min function.